[hardknott] Display the CPLD reset_source to users during boot.

amstewart opened the pull request:
We have had recent customer escalations where knowing the reset source on a cRIO controller has been critical. This information is available on most Linux RT cRIOs by reading the appropriate CPLD register. It was useful for customer troubleshooting to create a log that recorded this reset source at system startup. Having this kind of logging implemented as a feature could make troubleshooting unexpected system reboots much easier since we'll have a record of previous reset sources and can narrow down root causes quickly from there. 

This patchset adds a new initscript to the `ni-rtfeatures` package - appropriately called `/etc/init.d/ni-rtfeatures`. The new initscript reads the `/dev/rtfeatures/reset_source` sysfs entry, interprets the value, and then prints its interpretation to the boot log, and to the message log.

This patchset also generally cleans up the `ni-rtfeatures` recipe to conform to modern styling.

**Example Console Output**
```
Starting NIAuth: done                                                           
Starting OpenBSD Secure Shell server: sshd                                      
done.                                                                           
Starting Advanced Configuration and Power Interface daemon: acpid.              
<46>Aug 22 16:23:33 ni-rtfeatures: reset_source=processor  # Reset from MAX or command line
Starting NI Service Locator: done                                               
```
**Example /var/log/messages Output**
```
2022-08-22T16:23:33.000+00:00 NI-cRIO-9042-01E4AFBF ni-rtfeatures: reset_source=processor  # Reset from MAX or command line                                 
```

Natinst AZDO: https://dev.azure.com/ni/DevCentral/_workitems/edit/1004889

# Testing
* [x] Installed this package to a 9042 and verified that the `reset_source` is now printed to both console and to `/var/log/messages`.
* [x] Verified the same behavior on a crio-903x.
* [ ] Verified the same behavior on a crio-905x.
* [x] Tested the script output with the various possible (and impossible) values of `reset_source`.

amstewart committed:
ni-rtfeatures: add ni-rtfeatures initscript

The nirtfeatures driver writes a sysfs entry "reset_source" which
contains an informational string describing why the CPLD was reset. This
information is generally useful to NILRT users, but is not advertised.

Create an initscript - /etc/init.d/ni-rtfeatures - which reads this
sysfs and prints an informative message during boot, and to the syslog.

Add this initscript to the recipe.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-rtfeatures: remove PV from filename

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-rtfeatures: syntax and styling

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-rtfeatures: do not start initscripts at install

Neither the handle_cpld_ip_reset nor ni-rtfeatures initscripts should be
run after boot has already completed.

Remove the `-s` option from update-rc.d when running on a live system.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart commented:
I'll post samples of what the output looks like on Monday.

amstewart commented on recipes-ni/ni-rtfeatures/ni-rtfeatures.bb:
```diff
@@ -0,0 +1,66 @@
+	else
+		OPT=""
+	fi
+	update-rc.d $OPT handle_cpld_ip_reset start 6 1 3 4 5 .
+	update-rc.d $OPT ni-rtfeatures start 20 1 3 4 5 .
```
I would have rolled these scripts into one, but there is a race issue with syslog. `handle_cpld_ip_reset` wants to start before `sshd` - so <09. `ni-rtfeatures` needs to start after `syslog` - so >19. I thought for a bit about potentially moving sshd upward, or syslog downward, but eventually decided against it.

But note to future devs: that is why this initscript is split.

amstewart commented on recipes-ni/ni-rtfeatures/files/ni-rtfeatures.initd:
```diff
@@ -0,0 +1,70 @@
+log () {
+	local level=$1
+	local msg=${@:2}
+	local logger_stderr=""
+	[ "$VERBOSE" ] && logger_stderr="--stderr"
```
@SparkingSpork 
Is it desirable to hide the console output if the user doesn't have `VERBOSE=yes` during boot? Or should we show it all the time?

SparkingSpork replied:
I think showing this all the time is reasonable. It's a single line and is useful information regardless of if there's detailed debugging going on or not.

amstewart replied:
Removed the unused `logger_stderr` logic in latest commits.

amstewart committed:
ni-rtfeatures: remove initscripts in postrm

Package initscripts should be removed in the postrm, not the prerm.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-rtfeatures: remove usused VERBOSE logic

The ni-rtfeatures initscript output is always useful to users, so we
always show the output to the user on the console via the `logger
--stderr` option.

Remove the unused code which might have switched that logic when VERBOSE
is defined.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-rtfeatures: remove PID from logger output

The `--id` option is supposed to add the PID of the calling script to
the logger output like...

<46>Aug 22 15:16:20 ni-rtfeatures[1462]: reset_source=processor  # Reset from MAX or command line

But since `nirtfeatures` isn't a daemon, that PID isn't useful. The
initscript drops off of the process table immediately after.

Remove the option from the `logger` call.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
